vars/multicloudDeploy.groovy

```diff
@@ -99,16 +99,17 @@ void call(parameters = [:]) {
         if (config.cfTargets) {
 
             def deploymentType = DeploymentType.selectFor(CloudPlatform.CLOUD_FOUNDRY, config.enableZeroDowntimeDeployment).toString()
-            def deployTool = script.commonPipelineEnvironment.configuration.isMta ? 'mtaDeployPlugin' : 'cf_native'
-
-            // An isolated workspace is only required when using blue-green deployment with multiple cfTargets,
-            // since the cloudFoundryDeploy step might edit the manifest.yml file in that case.
-            Boolean runInIsolatedWorkspace = config.cfTargets.size() > 1 && deploymentType == "blue-green"
 
             for (int i = 0; i < config.cfTargets.size(); i++) {
 
                 def target = config.cfTargets[i]
 
+                // An isolated workspace is required when using blue-green deployment with multiple cfTargets,
+                // since the cloudFoundryDeploy step might edit the manifest.yml file in that case.
+                // It is also required in case of parallel execution and use of mtaExtensionCredentials, since the
+                // credentials are inserted in the mtaExtensionDescriptor file.
+                Boolean runInIsolatedWorkspace = config.cfTargets.size() > 1 && (deploymentType == "blue-green" || (target.mtaExtensionCredentials && config.parallelExecution))
```

Comment:
I see why you moved it here. But this basically means that for deployments we isolate and from some not. 
Also what is if mtaExtensionCredentials is defined in the cloudFoundryDeploy step and not here?

Reply:
Oh yes I see the problem. The only solution I can think of at the moment would be to change the condition like that, i.e., isolate all parallel deployments: 
```suggestion
                Boolean runInIsolatedWorkspace = config.cfTargets.size() > 1 && (deploymentType == "blue-green" || config.parallelExecution)
```